src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -236,6 +243,34 @@ public SignatureTrustAndValidityVerificationProvider()
                 timestampSummary.Issues.Concat(status.Issues));
         }
 
+        private bool IsUntrustedRootAllowed(Signature signature)
+        {
+            if (_allowUntrustedRootList != null)
+            {
+                foreach (var allowUntrustedRoot in _allowUntrustedRootList)
```

Comment:
so this `allowedUntrustedRoot` is applicable for all the certificates in a signature?

Reply:
A signature has a signing certificate, this can chain to a trusted root or not. In our case, a package can have a signature and a countersignature. This list provides certificates that are allowed to chain to an untrusted root, therefore we need to compare the signing certificate for the signature and the countersignature (if present and needed).